File: feature_2_extensions/Example2.java
package feature_2_extensions;

public class Example2 {

    public static void main(String[] args) {
        String example = "Hello, World!";

        System.out.println("Original: " + example);
        System.out.println("Reversed: " + StringExtensions.reverse(example));

        String palindrome = "racecar";
        System.out.println("Is " + palindrome + " a palindrome? " + StringExtensions.isPalindrome(palindrome));
        System.out.println("Original: " + palindrome);
        System.out.println("Reversed: " + StringExtensions.reverse(palindrome));
    }
}

File: feature_2_extensions/example2.dart
import 'string_extensions.dart';

void main(List<String> args) {
  String example = 'Hello World!';

  print('Original: $example');
  print('Reversed: ${example.reverse()}');

  String palindrome = 'racecar';
  print('Is $palindrome a palindrome? ${palindrome.isPalindrome()}');
  print('Original: $palindrome');
  print('Reversed: ${palindrome.reverse()}');
}
